### DISCUSSION TITLE
@layer components does not work

### DISCUSSION THREAD

**Discussion by shahul01** [tailwindlabs/tailwindcss#6263] [Help] [answered]
Everything works in Tailwind except @layer...
Here are the codes : 

HTML / Svelte:

```
.component {
    @apply egfcc;
}
```

CSS:

```
@tailwind base;
@tailwind components;
@tailwind utilities;

@layer components {
  .egfcc {
    @apply flex justify-center items-center;
  }

}
```

Simple code but does not work, tried restarting and hard refreshing, nothing works.
 
Throws error: `The egfcc class does not exist. If egfcc is a custom class, make sure it is defined within a @layer directive.`

I also added `postcss-import` plugin in postcss.config.js file as suggested by @adamwathan in another post just in case.

also added the class to purge safelist but nothing seems to work and there is no answers anywhere...

<br />

## Edit: SOLVED 

Implementation:
```
// tailwind.config.js

   
  plugins: [
    plugin(
      function({ addUtilities, addComponents, e, prefix, config }) {

        const myComponents = {

          '.shfcs': {
            'display': 'flex',
            'justify-content': 'center',
            'align-items': 'start',
          },



        }


        addComponents(myComponents)
      }
    ),

  ],
```

**Comment by adamwathan** [comment, marked as answer]
The problem is Svelte processes all of your CSS files in complete isolation, so the file where you are trying to apply that class is using a different instance of Tailwind than the file where that class is defined.

The easiest solution is to define any classes you want to apply across files using Tailwind’s plugin system instead of using CSS. Unfortunately nothing we can do about it because of how these build tools have decided to handle per component CSS. Same problem exists in Vue projects for example.

**Reply by adamwathan** [reply]
Here’s a previous/related discussion: https://github.com/tailwindlabs/tailwindcss/discussions/5509#discussioncomment-1333703

**Reply by shahul01** [reply]
Thank you @adamwathan for the answer!
This solution seems to work.

I searched everywhere and this is the only solution that works.
